### PR TITLE
Add JSONC support for JSON Plugin

### PR DESCRIPTION
**Pull request by kwaa**
## Description

JSONC format support, reusing existing dependencies.
I tried to write some tests... But it seems that many runs fail

### Check List

- [x] Have you read the
      [CODE OF CONDUCT](https://github.com/lumeland/lume/blob/master/CODE_OF_CONDUCT.md)
- [x] Have you read the document
      [CONTRIBUTING](https://github.com/lumeland/lume/blob/master/CONTRIBUTING.md)
  - [x] One pull request per feature. If you want to do more than one thing,
        send multiple pull request.
  - [x] Write tests.
  - [x] Run deno `fmt` to fix the code format before commit.
  - [x] Document any change in the `CHANGELOG.md`.

### CODE REVIEW

File: core/loaders/json.ts [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 import { isPlainObject, read } from "../utils.ts";
 
 import type { Data } from "../../core.ts";
 
-/** Load and parse a JSON file */
+/** Load and parse a JSON / JSONC file */
 export default async function json(path: string): Promise<Data> {
   const text = await read(path, false);
-  const content = JSON.parse(text);
+  const content = parse(text);
```
Comment (oscarotero) [inline core/loaders/json.ts:9]: For performance reasons, I'd use jsonc only for files with the jsonc extension:

```js
const content = path.endsWith(".jsonc") ? parse(text) : JSON.parse(text);
```

**Review by oscarotero — CHANGES_REQUESTED** [review]

File: core/loaders/json.ts (hunk 1 above)
Comment (kwaa) [reply core/loaders/json.ts:9]: > For performance reasons, I'd use jsonc only for files with the jsonc extension:

OK, it has been updated.

**Comment by kwaa** [comment]
Successful completion of the test...
If needed, I can also add a new `jsonc.test.ts` by referring to `toml.test.ts`.

**Comment by kwaa** [comment]
> If needed, I can also add a new `jsonc.test.ts` by referring to `toml.test.ts`.

Considering that it is part of the JSON Plugin, I ended up adding a `json.test.ts`

**Comment by oscarotero** [comment]
Okay, thanks!

**Pull request merged**